lib/src/controller.dart

```diff
@@ -20,23 +20,22 @@ class TimetableController<E extends Event> {
     this.visibleRange = const VisibleRange.week(),
     this.firstDayOfWeek = DayOfWeek.monday,
   })  : assert(eventProvider != null),
-        initialDate = initialDate ?? LocalDate.today(),
+        assert(visibleRange.dateInsideAvailableRange(initialDate ?? LocalDate.today())),
+        initialDate = visibleRange.getPeriodStartDate(initialDate ?? LocalDate.today(), firstDayOfWeek),
         assert(initialTimeRange != null),
         assert(firstDayOfWeek != null),
         assert(visibleRange != null) {
     _scrollControllers = LinkedScrollControllerGroup(
-      initialPage: visibleRange.getTargetPageForFocusDate(
-          this.initialDate, firstDayOfWeek),
       viewportFraction: 1 / visibleRange.visibleDays,
     );
 
     _dateListenable = scrollControllers.pageListenable
-        .map((page) => LocalDate.fromEpochDay(page.floor()));
+        .map((page) => this.initialDate.addDays(page.floor()));
     _currentlyVisibleDatesListenable = scrollControllers.pageListenable
         .map((page) {
       return DateInterval(
-        LocalDate.fromEpochDay(page.floor()),
-        LocalDate.fromEpochDay(page.ceil() + visibleRange.visibleDays - 1),
+        this.initialDate.addDays(page.floor()),
```

Comment:
Is the meaning of `page` now: offset from `initialDate`? That would be a breaking change and hence something I'd like to avoid

Reply:
basically yes, page (which now can be infinite in both positive and negative directions) is offset for the initial date. It means that the initial date can be changed as well if needed, obviously, it will require some additional work for all-day  event Widget)) but overall - having page as an offset of initialDate simplifies some calculations.

One particular stuff that I don't like is the initial date that the user passed into the controller, and the initial date that will be actually defined can be different in the next cases, that passed initialDate can't be actually initial date. For example:
- it's Week visible range, and initial date, that user passed in, not the first day of the week
- the visible range was defined for example as 5 days with some end date, and the initial date - not the start date of this range

Like max date 10th, range is 5 days, and the initial date was specified as 8th - so it can't be initial date, since range should start from 6th date

Reply:
as for breaking change - doubt that anyone was relying on page indexes like on epochDays value.

Reply:
btw, having page as an offset against initial page potentially should simplify modularization, since in that case dev won't need to take into account all this internal specific of scroll list building

Reply:
Even if we don't know for sure if someone is using these indices to calculate dates, this _is_ a behavioral change of the public API and hence requires a breaking change. I don't think it's worth the hassle for all users of this package to manually update the package due to this relatively minor modification. Could you please work with the existing `page` values?

Regarding `initialDate`: This also seems to be a breaking change if someone is using a `WeekVisibleRange`…

Reply:
> Could you please work with the existing page values?

You mean return back approach where Page is actually days since the epoch? Or add shift to pageListenable object?